scipy/stats/_axis_nan_policy.py

```diff
@@ -509,6 +583,10 @@ def hypotest_fun(x):
 
             x = np.moveaxis(x, axis, -1)
             res = np.apply_along_axis(hypotest_fun, axis=-1, arr=x)
+            if keepdims:
+                return _add_reduced_axes(result_unpacker(res),
```

Comment:
This is still a return.

Reply:
When `keepdims` is True, the `result_unpacker` is applied (to get a tuple of results) and then the `result_object` before returning from `_add_reduced_axes`. The default `result_unpacker` doesn't work when the result is an instance of the `result_object` (so, the line below this one `return result_object(*result_unpacker(res))` fails). See https://github.com/scipy/scipy/pull/15478#discussion_r795037172

This is why it is tricky to make it work without the `if keepdims` before it and returning from there. I think we can avoid applying the result_object in the `_add_reduced_axes` function but I thought it might make things a little more confusing and unreadable.

Reply:
Merged the two return statements in [latest commits](https://github.com/scipy/scipy/pull/15478/commits/54f0b09bc953a3deadaa56245cee4007e7ab66e5).